### PR TITLE
Date Filter

### PR DESCRIPTION
**Pull request by julieflament**
- added filter which filters the hurricanes based on month and year
- initially shows all of the hurricanes until a filter is used
- updates the heatmap after user inputs what values they want in filter
- completed #100

### CODE REVIEW

**Review by noahstaz — CHANGES_REQUESTED** [review]
Code looks, good, but a test file should be added to ensure that everything works properly, also merge conflicts

File: code/app.py [hunk 1]
```diff
@@ -128,8 +76,151 @@ def alerts_page():
...
+@app.route('/delete-user/<int:user_id>')
+def delete_user(user_id):
+    try:
+        conn = sqlite3.connect(os.path.join(basedir, 'hurriscan.db'))
+        cur = conn.cursor()
+        cur.execute('DELETE FROM User WHERE id = ?', (user_id,))
+        conn.commit()
+        
+        conn.close()
+        return redirect('/admin') # Doesn't do anything, but needed to return something ¯\_(ツ)_/¯
```
Comment (noahstaz) [inline code/app.py:122]: Maybe try to return null or something else instead

File: code/app.py [hunk 2]
```diff
@@ -128,8 +76,151 @@ def alerts_page():
...
+        filtered_data = cursor.fetchall()
+        connection.close()
+
+    return jsonify(filtered_data)
+
+
+    return jsonify(filtered_data)
 if __name__ == '__main__':
     app.run(debug=True)
+
```
Comment (noahstaz) [inline code/app.py:226]: Remove the extra space here

File: code/app.py (hunk 1 above)
Comment (julieflament) [reply code/app.py:122]: thats not me thats trevor i think so I'm not touching that lol

File: code/templates/mapfilter-temp/mapfilter.html [hunk 3]
```diff
@@ -112,6 +95,72 @@
...
+        .then(response => response.json())
+        .then(data => {
+            // Update map with filtered data
+        })
+        .catch(error => {
+            console.error('Error fetching filtered data:', error);
+        });
+}
+
+
```
Comment (noahstaz) [inline code/templates/mapfilter-temp/mapfilter.html:160]: Also remove this extra spacing

File: code/app.py (hunk 1 above)
Comment (noahstaz) [reply code/app.py:122]: Ah, @trevorwinser

Comment (trevorwinser) [reply code/app.py:122]: I believe I tried null and it was upset with null returns.

**Review by trevorwinser — CHANGES_REQUESTED** [review]
map-filter:125 Uncaught SyntaxError: Unexpected token '='
3map-filter:24 Uncaught ReferenceError: applyDateFilter is not defined
    at HTMLButtonElement.onclick (map-filter:24:41)
2map-filter:30 Uncaught ReferenceError: applyTemperatureRangeFilter is not defined
    at HTMLButtonElement.onclick (map-filter:30:53)

Error occur when attempting to run. Map does not display.

**Review by trevorwinser — APPROVED** [review]
Everything works fine after bug fix!

**Review by noahstaz — APPROVED** [review]
Looks good fixed some minor things and created #136 to address testing findings, one the merge conflicts are resolved good to merge

**Pull request merged**